### PR TITLE
[12.x] chore: remove support for Carbon v2

### PR DESCRIPTION
**Pull request by calebdw**
Hello!

This removes support for Carbon v2 and cleans up the `addReal*` methods to use the `add*` equivalents.

Thanks!

### CODE REVIEW

**Comment by crynobone** [comment]
`addReal*` was deprecated in Carbon 3.2 but should still work: https://github.com/briannesbitt/Carbon/commit/a6e64c765cefc7f42922db9cfa11b209cb67677c

It might be best to target `master` branch for v12 and remove support for Carbon 2

**Comment by calebdw** [comment]
Ah, PHPStan was complaining about the methods not existing and I missed the dynamic `addRealUnit` method

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -64,15 +64,6 @@ jobs:
...
 
-      - name: Set Minimum PHP 8.4 Versions
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 5
-          max_attempts: 5
-          command: composer require  nesbot/carbon:^3.4 --no-interaction --no-update
-          shell: bash
-        if: matrix.php >= 8.4
-
```
Comment (crynobone) [inline .github/workflows/tests.yml:75]: Can you revert these changes

**Comment by crynobone** [comment]
Thanks @kylekatarnls
